src/Billing/Program.cs

```diff
@@ -9,59 +9,75 @@
 
 var title = string.IsNullOrEmpty(instancePostfix) ? "Failure rate (Billing)" : $"Billing - {instancePostfix}";
 var instanceName = string.IsNullOrEmpty(instancePostfix) ? "billing" : $"billing-{instancePostfix}";
-
 var instanceId = DeterministicGuid.Create("Billing", instanceName);
+var prometheusPortString = args.Skip(1).FirstOrDefault();
 
-var endpointConfiguration = new EndpointConfiguration("Billing");
-endpointConfiguration.LimitMessageProcessingConcurrencyTo(4);
+var endpointControls = new ProcessingEndpointControls(() => PrepareEndpointConfiguration(instanceId, instanceName, prometheusPortString));
 
-var serializer = endpointConfiguration.UseSerialization<SystemJsonSerializer>();
-serializer.Options(new JsonSerializerOptions
-{
-    TypeInfoResolverChain =
-        {
-            MessagesSerializationContext.Default
-        }
-});
+var ui = new UserInterface();
+endpointControls.BindSlowProcessingDial(ui, '5', 't');
+endpointControls.BindDatabaseFailuresDial(ui, '6', 'y');
+
+endpointControls.BindDatabaseDownToggle(ui, 'f');
+endpointControls.BindDelayedRetriesToggle(ui, 'g');
+endpointControls.BindAutoThrottleToggle(ui, 'h');
 
-var transport = new LearningTransport
+endpointControls.BindFailureReceivingButton(ui, 'v');
+endpointControls.BindFailureProcessingButton(ui, 'b');
+endpointControls.BindFailureDispatchingButton(ui, 'n');
+
+if (prometheusPortString != null)
 {
-    StorageDirectory = Path.Combine(Directory.GetParent(Assembly.GetExecutingAssembly().Location)!.Parent!.FullName, ".learningtransport"),
-    TransportTransactionMode = TransportTransactionMode.ReceiveOnly
-};
-endpointConfiguration.UseTransport(transport);
+    OpenTelemetryUtils.ConfigureOpenTelemetry("Billing", instanceId.ToString(), int.Parse(prometheusPortString));
```

Comment:
This doesn't prevent the demo from running without Prometheus right?

Reply:
Yes, it just exposes the data.